util/log/log.go

```diff
@@ -34,74 +43,169 @@ func FatalOnPanic() {
 	}
 }
 
+func logKV(buf *bytes.Buffer, kvs ...[]interface{}) {
+	var i int
+	var s string
+	var kv []interface{}
+	for kvi := range kvs {
+		kv = kvs[kvi]
+		l := len(kv)
+		for i = 0; i < l; i++ {
+			if i%2 == 0 {
+				if kvi != 0 || i != 0 {
+					buf.WriteString(" ")
+				}
+			} else {
+				buf.WriteString("=")
+			}
+			switch kt := kv[i].(type) {
+			case string:
+				s = kt
+			case fmt.Stringer:
+				s = kt.String()
+			default:
+				s = fmt.Sprintf("%v", kt)
+			}
+			if i%2 == 0 {
+				buf.WriteString(s)
+			} else {
+				buf.WriteString(strconv.Quote(s))
+			}
+		}
+		if i%2 != 0 {
```

Comment:
yeah, I'll move this up. And for the panic, I see what you're saying, and I'd hate for this to crash a production system on a rarely-executed code path with a wrong log call, but I think for now it'll do more good than bad just shooting it in the head. I'll add a TODO though.